src/HttpFactory.php

```diff
@@ -83,11 +95,17 @@ public function createServerRequest(string $method, $uri, array $serverParams =
         return new ServerRequest($method, $uri, [], null, '1.1', $serverParams);
     }
 
+    /**
+     * @throws \InvalidArgumentException
+     */
     public function createResponse(int $code = 200, string $reasonPhrase = ''): ResponseInterface
     {
         return new Response($code, [], null, '1.1', $reasonPhrase);
     }
 
+    /**
+     * @throws \InvalidArgumentException
```

Comment:
If this throws doc is not present on the psr interface, we are actually violating the contract here.

Reply:
I see. 2 possible solutions that come in my mind:
- Add `@throws` to the psr interface
- Do not add `@throws` here and add phpstan error to baseline so check passes.

Anyways as i would understand the contract, it says that `\Psr\Http\Message\RequestFactoryInterface::createRequest()` can never fail and will aways return instance of `RequestInterface`, which i believe is not true

Reply:
I'm not saying your doc is wrong. :P As you say, either the psr interface is missing some information, or our implementation is bad. @Nyholm what do you think here?